src/commit.c

```diff
@@ -350,7 +350,11 @@ int apk_solver_commit_changeset(struct apk_database *db,
 
 	if (!db->performing_self_upgrade) {
 		if (errors)
-			snprintf(buf, sizeof(buf), "%d errors;", errors);
+			if (errors != 1)
+				snprintf(buf, sizeof(buf), "%d errors;", errors);
+			else
+				snprintf(buf, sizeof(buf), "%d error;", errors);
```

Comment:
On modern optimizing compiler the generated code should be equivalent. If simplification is requested the string could be just `%d error(s);`. Alternatively to reduce the string literal sizes we could do
```c
snprintf(buf, sizeof buf, "%d error%s;", errors, errors ? "s" : "");
```
but any saving on string literal size is probably wasted on pushing the extra argument. So I guess we can just pick the one that looks most readable and compact.